File: yt_assistant_client/src/utils/logger.ts
import pino from "pino";

const isDev =
  import.meta.env.VITE_ENV === "local" || import.meta.env.VITE_ENV === "dev";
const isTest = import.meta.env.VITE_ENV === "test";

export const logger = pino({
  level: isTest ? "silent" : isDev ? "debug" : "info",
  transport: isDev
    ? {
        target: "pino-pretty",
        options: {
          colorize: true,
        },
      }
    : undefined, // no transport (default transport) in prod and test
});

File: yt_assistant_api/core/logger.py
import logging
import os
import sys

from .constants import Env

# Determine log level
log_levels = {
    Env.LOCAL: logging.DEBUG,
    Env.DEV: logging.INFO,
    Env.TEST: logging.WARNING,
    Env.PROD: logging.ERROR,
}
env = Env[os.getenv("ENV").upper()]
log_level = log_levels[env]

# Create logger
logger = logging.getLogger()
formatter = logging.Formatter(fmt="%(asctime)s - %(levelname)s - %(message)s")
stream_handler = logging.StreamHandler(sys.stdout)
stream_handler.setFormatter(formatter)
# file_handler = logging.FileHandler("app.log")
# file_handler.setFormatter(formatter)
logger.handlers = [
    stream_handler,
]
logger.setLevel(log_level)
